authorize/grpc.go

```diff
@@ -331,3 +334,13 @@ func logAuthorizeCheck(
 
 	evt.Msg("authorize check")
 }
+
+func replaceCheckRequestURL(req *envoy_service_auth_v2.CheckRequest, newURL *url.URL) {
+	req.Attributes.Request.Http.Scheme = newURL.Scheme
+	req.Attributes.Request.Http.Host = newURL.Host
+	// the go path is decoded, but the check request path is encoded, so we need to re-encode it
+	req.Attributes.Request.Http.Path = newURL.EscapedPath()
```

Comment:
Maybe we can instead use `X-Original-URL` per other comment if set to grab the unencoded path?